File: js/src/formulas/matrixMultiply.ts
export default function matrixMultiply (matrixA: number[][], matrixB: number[][]) {
  if (matrixA[0].length !== matrixB.length) {
    throw new Error()
  }
  const result = [] as number[][]
  for (let i = 0; i < matrixA.length; i++) {
    result.push([])
    for (let j = 0; j < matrixB[0].length; j++) {
      let sum = 0
      for (let k = 0; k < matrixB.length; k++) {
        if (!isNaN(matrixA[i][k]) && !isNaN(matrixB[k][j])) {
          sum += matrixA[i][k] * matrixB[k][j]
        }
      }
      result[i].push(sum)
    }
  }

  return result
}

File: py/src/formulas/matrixMultiply.py
def matrix_multiply(matrixA: list[list[float]], matrixB: list[list[float]]) -> list[list[float]]:
    if len(matrixA[0]) != len(matrixB):
        raise Exception()
    result = []
    for i in range(len(matrixA)):
        result.append([])
        for j in range(len(matrixB[0])):
            sum_ = 0
            for k in range(len(matrixB)):
                if not (matrixA[i][k] != matrixA[i][k] or matrixB[k][j] != matrixB[k][j]):  # not NaN
                    sum_ += matrixA[i][k] * matrixB[k][j]
            result[i].append(sum_)
    return result
